nasa-gibs/worldview
# WV-2181

## Description

Fixes:  Right click then Take a snapshot causes error.  

When right clicking on the map, coordinates are set in the Redux store.  When single-clicked off of the right-click menu and not selecting anything,  coordinates were still stored.  In `web/js/map/ui.js` there is a function `subscribeToStore` that runs `addMarkerAndUpdatedStore()` when the `TOGGLE_DIALOG_VISIBLE` is called.  This action is called when the snapshot button is clicked is it trying to close the little tooltip next to the location marker.  Since a location marker doesn't exist yet, it was throwing an error.  The new boolean `isMarkerPlaced` in `locationSearch` was added to show declare whether a marker exists on the map or not making the markers independent from coordinates existing in the store.

## How To Test

### Test 1
1. Pull down changes and run `npm start` and open [localhost:3000](http://localhost:3000).
2. Right click on map to bring up context menu, but don't select anything in the menu.
3. Click on "Take a Snapshot"
4. No error should show.  The Snapshot menu should open with the grayed out screen and a non-grayed out selection box.

### Test 2
1. Right click on map to bring up context menu, and place a marker on the map.
1. Right click again and do NOT select anything, leaving the menu open.
1. Click on "Take a Snapshot"
1. No error should show.  The Snapshot menu should open with the grayed out screen and a non-grayed out selection box.
1. The location marker's tooltip should minimize/disappear and the marker should be by itself.
